reconcile/ocm_machine_pools.py

```diff
@@ -330,12 +357,24 @@ def fetch_current_state(
     }
 
 
-def _is_hypershift(cluster: ClusterV1) -> bool:
-    return bool(cluster.spec and cluster.spec.hypershift)
+def _classify_cluster_type(cluster: ClusterV1) -> ClusterType:
+    assert cluster.spec  # cluster.spec is required
```

Comment:
Ist that assert called? Wouldn't it be nicer to throw a ValueError with some description?

Reply:
right, somehow this field is not required in schema, I'll do a explicit check here